Assets/MixedRealityToolkit/Utilities/MixedRealityServiceRegistry.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+    /// <summary>
+    /// Static class that represents the Mixed Reality Toolkit service registry.
+    /// </summary>
+    /// <remarks>
+    /// The service registry is used to enable discovery of and access to active Mixed Reality Toolkit services at
+    /// runtime without requiring direct code reference to a singleton style component.
+    /// </remarks>
+    public static class MixedRealityServiceRegistry
+    {
+        // todo: is this the data type we want?
+        /// <summary>
+        /// The service registry store where the key is the Type of the service interface and the value is
+        /// a pair in which they key is the service instance and the value is the regisrar instance.
```

Comment:
>pair [](start = 14, length = 4)

I think this is a list of pairs - TBH it feels like this datastructure is interesting enough to have its own class abstraction. Anytime you start going 3 layers deep (i.e. a hashtable of a list of pairs) it probably means it should be abstracted a little different.

Reply:
Willing to offline chat about your thoughts? We can update here with the result.

Reply:
we discussed internally and decided to leave as is